src/storage/storage.c

```diff
@@ -474,18 +489,20 @@ storageInfoList(
                 .storage = this,
                 .callbackFunction = callback,
                 .callbackData = callbackData,
+                .expression = param.expression,
                 .sortOrder = param.sortOrder,
                 .recurse = param.recurse,
                 .path = path,
             };
 
-            if (param.expression != NULL)
-                data.expression = regExpNew(param.expression);
+            if (data.expression != NULL)
+                data.regExp = regExpNew(param.expression);
```

Comment:
This is redundant for S3 but not for Posix. In s3 we pass the expression and then in storageS3ListInternal we create the regExp again there instead. So by the time we get back to storageInfoListCallback is it meaningless to apply the regExp because in s3 that was handled at the driver level but in Posix it was not.

Reply:
Yeah, it should not have been fully applied in S3. This has been fixed. See comment above.